### PR TITLE
Passing labels to text_decoder to compute loss.

### PR DESCRIPTION
**Pull request by kapulkin**
I noticed, that labels variable is not passed to text_decoder in VLMForCausalLM.forward(). So text_decoder will return just logits and will not compute loss. This makes impossible to use VLMForCausalLM model with transformer.Trainer and requires to write custom train loop or wrap VLMForCausalLM.

There is a fix to avoid that incompatibility.

### CODE REVIEW

**Comment by kimihailv** [comment]
Passing labels to the text decoder is not enough. `input embeds` contain not only embeddings of text tokens, but also image features, so logits will also contain not only data for text but also for image
